### PR TITLE
Companion: Rename pallet trait `Trait` to `Config`

### PR DESCRIPTION
**Pull request by thiolliere**
This PR renames the pallet trait Trait to Config.

Config storage in Configuration pallet is also renamed ActiveConfig. This doesn't need a migration as Configuration pallet is not used yet. (https://github.com/paritytech/polkadot/pull/2036 )

companion of https://github.com/paritytech/substrate/pull/7599

### CODE REVIEW

File: runtime/parachains/Cargo.toml [hunk 1]
```diff
@@ -12,27 +12,27 @@ rustc-hex = { version = "2.1.0", default-features = false }
...
-sp-api = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-inherents = { package = "sp-inherents", git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-std = { package = "sp-std", git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-io = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-session = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-staking = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-sp-keystore = { git = "https://github.com/paritytech/substrate", branch = "master", optional = true }
+sp-api = { git = "https://github.com/paritytech/substrate", default-features = false , branch = "gui-rename-trait-to-config" }
```
Comment (ordian) [inline runtime/parachains/Cargo.toml:15]: are these intentional?

Comment (thiolliere) [reply runtime/parachains/Cargo.toml]: ah no this is reverted now

File: runtime/parachains/src/configuration.rs [hunk 2]
```diff
@@ -114,24 +114,24 @@ pub struct HostConfiguration<BlockNumber> {
...
 
-pub trait Trait: frame_system::Trait { }
+pub trait Config: frame_system::Config { }
 
 decl_storage! {
-	trait Store for Module<T: Trait> as Configuration {
+	trait Store for Module<T: Config> as Configuration {
 		/// The active configuration for the current session.
-		Config get(fn config) config(): HostConfiguration<T::BlockNumber>;
+		HostConfig get(fn config) config(): HostConfiguration<T::BlockNumber>;
```
Comment (thiolliere) [inline runtime/parachains/src/configuration.rs:122]: note for reviewer: this conflicted thus I changed to HostConfig

Comment (thiolliere) [reply runtime/parachains/src/configuration.rs:122]: I guess this pallet is used in some test chains, thus it needs to be migrated, I open such migration/rename in an independant PR for ease reviewing. https://github.com/paritytech/polkadot/pull/2036

**Review by ordian — APPROVED** [review]
Looks good, assuming this compiles fine.
No need to worry about breaking rococo.

**Comment by ordian** [comment]
Looks like #2038 sneaked in causing a merge conflict.

**Comment by thiolliere** [comment]
bot merge

**Review by kianenigma — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**